internal/services/communication/communication_service_data_source.go

```diff
@@ -0,0 +1,146 @@
+	SecondaryKey              string            `tfschema:"secondary_key"`
+	Tags                      map[string]string `tfschema:"tags"`
+}
+
+func (CommunicationServiceDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.CommunicationServiceName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+	}
+}
+
+func (CommunicationServiceDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"data_location": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"primary_connection_string": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"primary_key": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"secondary_connection_string": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"secondary_key": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"tags": commonschema.TagsDataSource(),
+	}
+}
+
+func (CommunicationServiceDataSource) ModelObject() interface{} {
+	return &CommunicationServiceDataSourceModel{}
+}
+
+func (CommunicationServiceDataSource) ResourceType() string {
+	return "azurerm_communication_service"
+}
+
+func (CommunicationServiceDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Communication.ServiceClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var state CommunicationServiceDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := communicationservices.NewCommunicationServiceID(subscriptionId, state.ResourceGroupName, state.Name)
+
+			resp, err := client.Get(ctx, id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s was not found", id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			keysResp, err := client.ListKeys(ctx, id)
+			if err != nil {
+				return fmt.Errorf("listing keys for %s: %+v", id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				if props := model.Properties; props != nil {
+					state.DataLocation = props.DataLocation
+				}
+
+				if model.Tags != nil {
+					state.Tags = *model.Tags
+				}
+			}
+
+			if model := keysResp.Model; model != nil {
+				if model.PrimaryConnectionString != nil {
+					state.PrimaryConnectionString = *model.PrimaryConnectionString
+				}
+
+				if model.PrimaryKey != nil {
+					state.PrimaryKey = *model.PrimaryKey
+				}
+
+				if model.SecondaryConnectionString != nil {
+					state.SecondaryConnectionString = *model.SecondaryConnectionString
+				}
+
+				if model.SecondaryKey != nil {
+					state.SecondaryKey = *model.SecondaryKey
+				}
```

Comment:
FWIW we can reduce this to:

```suggestion
				state.PrimaryConnectionString = pointer.From(model.PrimaryConnectionString)
				state.PrimaryKey = pointer.From(model.PrimaryKey)
				state.SecondaryConnectionString = pointer.From(model.SecondaryConnectionString)
				state.SecondaryKey = pointer.From(model.SecondaryKey)
```